extensions/pkg/util/shoot_clients.go

```diff
@@ -77,8 +77,9 @@ func NewClientForShoot(ctx context.Context, c client.Client, namespace string, o
 		err            error
 	)
 
-	// do we try the internal one first?
 	if os.Getenv("GARDENER_SHOOT_CLIENT") != "external" {
+		// First, we try to use the "internal" kubeconfig (the one with in-cluster address) as in-cluster traffic is free of charge.
+		// Otherwise, we fallback to the "external" kubeconfig (the one with external DNS name or load balancer address). It usually translates to egress traffic costs.
```

Comment:
Can you move this to the method docu comment a few lines above?
The env check is just to enforce always to use the external endpoint, ref #4366.

Reply:
Yes, but in general this method tries firstly the internal client and then the external due to the cost saving factor. 
The env variable is just a short cut to directly use the external client. And this again can be a documentation in the method that by setting the environment variable `GARDENER_SHOOT_CLIENT=external` the gardenlet will use only the external clients and will not try to use the internal one.

Reply:
> And this again can be a documentation in the method that by setting the environment variable GARDENER_SHOOT_CLIENT=external the gardenlet will use only the external clients and will not try to use the internal one.

I think it is already documented in the func doc string.

You can use github suggestion or share a git diff because I cannot follow.